pkg/kubelet/kubelet_pods.go

```diff
@@ -1521,6 +1527,20 @@ func getPhase(spec *v1.PodSpec, info []v1.ContainerStatus) v1.PodPhase {
 		// one container is running
 		return v1.PodRunning
 	case running == 0 && stopped > 0 && unknown == 0:
+		// Ensure all terminal pods are transitioned to a terminal phase.
+		// The pod is deleted so its containers won't be restarted regardless of
+		// the restart policy.
+		if podIsTerminal {
+			// TODO(#116484): Also assign terminal phase to static pods.
+			if !kubetypes.IsStaticPod(pod) && !kubetypes.IsMirrorPod(pod) {
```

Comment:
Why are you checking mirror pod here?  Mirror pods will never get passed to this function (if they do, that's very wrong).

Reply:
Done, checked locally and works fine.